mmv1/products/compute/api.yaml

```diff
@@ -1672,6 +1672,28 @@ objects:
         name: 'securityPolicy'
         description: |
           The security policy associated with this backend service.
+      - !ruby/object:Api::Type::NestedObject
+        name: 'securitySettings'
+        description: |
+          The security settings that apply to this backend service. This field is applicable to either
+          a regional backend service with the service_protocol set to HTTP, HTTPS, or HTTP2, and
+          load_balancing_scheme set to INTERNAL_MANAGED; or a global backend service with the
+          load_balancing_scheme set to INTERNAL_SELF_MANAGED.
+        properties:
```

Comment:
Are either of these fields required if the parent is set, or would the following message make sense in the API?

```
"securitySettings": {}
```

Reply:
Neither is required. Both are marked optional (in the internal comments/doc for the SecuritySettings config.). `"securitySettings": {}` is legal.

Reply:
It's desirable in Terraform that objects can't be specified like that- there's an unusual transformation of the value in Terraform's state, that causes a lot of issues (https://github.com/hashicorp/terraform-provider-google/issues/3928, the issue says single optional field but really means "where all fields are optional"). Giving it more thought, would it make sense to mark both fields as required? To not send the values to the API a user could do:

```
# intentionally empty - omitting the block altogether
```

or 

```
security_settings {
  client_tls_policy = ""
  subject_alt_names = []
} 
```

While the following wouldn't work:

```
security_settings {}
```

Reply:
Omitting the block altogether is probably the best way to send this when you have no values to send. Let me think about marking both required also CC @karthikbox

Reply:
Note- we still need to make a decision here. It seems like `clientTlsPolicy` should definitely be required, and `subjectAltNames` probably should be as well (a user can always just define an empty array)

Reply:
Both marked `required`